guide/src/class/object.md

```diff
@@ -52,14 +52,25 @@ contained inside `Number`.
 #[pymethods]
 impl Number {
     // For `__repr__` we want to return a string that Python code could use to recreate
-    // the `Number`, like `Number(5)` for example.
-    fn __repr__(&self) -> String {
+    // the `Number`, like `Number(5)` for example. If this is subclassed in Python we would
+    // like the class name to reflect the subclass name, so we'll use self.__class__.__name__
+    // to get the name of the class.
+    fn __repr__(self_: PyRef<Self>) -> PyResult<String> {
+        // We need both a reference to the Rust data and the Python data, so we use the 
+        // parameter type PyRef<Self>. This dereferences into Self but also can be converted
+        // into a PyObject.
+        let pyself = self_.clone().into_py(self_.py());
```

Comment:
Hmm does this only work if `Self` implements clone, meaning this is actually cloning the underlying struct? Is there a way to get a reference to the `PyObject` without moving `self_`?